bson/decoder.go

```diff
@@ -145,3 +164,34 @@ func (d *Decoder) DefaultDocumentM() {
 func (d *Decoder) DefaultDocumentD() {
 	d.defaultDocumentD = true
 }
+
+// AllowTruncatingDoubles causes the Decoder to truncate the fractional part of BSON "double" values
+// when attempting to unmarshal them into a Go integer (int, int8, int16, int32, or int64) struct
+// field. The truncation logic does not apply to BSON "decimal128" values.
+func (d *Decoder) AllowTruncatingDoubles() {
+	d.dc.Truncate = true
+}
```

Comment:
Consider adding a `bool` argument to the new option setters:

```go
func (d *Decoder) AllowTruncatingDoubles(value bool) {
	d.dc.Truncate = value
}
```

This allows disabling the option after it is enabled.

I am not sure if there is a compelling use case for disabling after enabling. Maybe one use case is to reuse a `Decoder` with one option changed?

The deprecated options, like `MinSize`, are exported fields and support disabling after enabling.

Reply:
>  Do you think adding an on/off boolean is a more usable API?

No. The suggestion was meant as an alternative to consider. I did not realize there was precedent with the Go JSON `Encoder`/`Decoder`.